distributed/dashboard/components/scheduler.py

```diff
@@ -2902,7 +2902,7 @@ def __init__(self, scheduler, **kwargs):
         y_range = Range1d(0, max(self.plugin.nthreads))
 
         self.root = figure(
-            id="bk-task-group-progress-plot",
+            # id="bk-task-group-progress-plot",
```

Comment:
A better alternative that should also work with 2.x is to set `name` instead of `id`. AFAIK that ill add a `name` class to the element that can be used for CSS selection. (cc @mattpap)

Reply:
Good to know. I tried that initially, but it looks like we're defining a `name` and `id` for some figures already (not sure why). I wanted to see if CI breaks if we just don't specify an `id` at all

Reply:
alternatively `css_classes` is still also an option 

https://docs.bokeh.org/en/dev-3.0/docs/reference/models/widgets/markups.html#bokeh.models.Div.css_classes

Reply:
@ian-r-rose do you have any insight here?

Reply:
I took a brief look through the dashboard codebase and didn't see anywhere where the DOM ids are explicitly used, so I think it should be relatively safe to just remove these `id=` parameters.

I *do* see a couple of places where bokeh components have custom CSS styling, and that will likely need some updating here. But that is a separate concern from the `id` parameters here.

Reply:
> I do see a couple of places where bokeh components have custom CSS styling, and that will likely need some updating here

Could you point me to those?

Reply:
I'm referring to these: https://github.com/dask/distributed/blob/5dccad4c0b719a57b53404b02e5666cf23b119fd/distributed/http/static/css/base.css#L100-L107

I'm not sure if they will work with Bokeh 3, but my read of the changelog is that the CSS system has been overhauled, so it may be worth verifying.

Reply:
@mattpap can you advise?

Reply:
Due to migration to shadow DOM (or web components, or modularized CSS; different names same idea), styling of bokeh's components with external style sheets is not possible anymore. Each component has to be styled individually. Many CSS classes are gone (e.g. `.bk-toolbar-box`) or renamed (simplified) (e.g. `.bk-toolbar-right` becomes `.bk-right`), because complex selectors are not needed anymore with modularized CSS. `.bk-root` is gone, because it's not needed anymore. Also the general structure of bokehjs components' DOM and CSS changed significantly, so even if custom CSS is updated to work with shadow DOM, it still will likely not do what's expected without additional changes.

If you have only individual (instances) components to style, then this should work (give or take, I didn't test it):

```css
.bk-root .bk-toolbar-box .bk-toolbar-right {
  top: 4px;
  right: 4px;
}
```

```py
Toolbar(stylesheets=["""
:host(.bk-right) {
  top: 4px;
  right: 4px;
}
"""])
```

```css
.bk-root .bk-data-table {
  z-index: 0;
}
```

```py
DataTable(stylesheets=["""
.bk-data-table {
  z-index: 0;
}
"""])
```

Most likely you don't need the second one, as we hopefully fixed all issues with overlapping components.

If you want to style all components of a given kind, like all toolbars, then we don't have a good replacement right now. It's possible to configure `stylesheets` property via a theme, but we are still working one something more robust. The current advice is that, if you have a lot of custom CSS, then probably you will have to postpone migration to 3.0 for a moment (until 3.0.n or 3.1; in particular in 3.1 we intend to define an "API" for bokehjs' CSS). If you have just a little bit of customization, then we can figure it out on case-by-case basis.